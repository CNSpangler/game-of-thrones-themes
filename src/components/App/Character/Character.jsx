import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, titles, aliases }) => {
  const aliasElements = aliases.map(alias => 
    <li key={alias.i}>alias</li>
  )
  
  return (
    <>
      Name: {name ? <h3>name</h3> : <h3>aliases[0]</h3>}
      Title: {titles && <h4>titles[0]</h4>}
      Aliases:
      <ul>{aliasElements}</ul>
    </>
  );};

Character.propTypes = {
  name: PropTypes.string,
  title: PropTypes.string,
  aliases: PropTypes.array
};

export default Character;
