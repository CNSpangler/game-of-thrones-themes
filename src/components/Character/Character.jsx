import React from 'react';
import PropTypes from 'prop-types';
import styles from '../App/App.css';

const Character = ({ name = 'unknown', culture, titles, aliases }) => {
  const aliasElements = aliases.map((alias, i) => 
    <li key={i}>{alias}</li>
  );
  
  return (
    <section className={styles.card}>
      {name ? <h3>Name: {name}</h3> : <h3>{aliases[0]}</h3>}
      {culture !== '' && <h4>Culture: {culture}</h4>}
      {titles[0] && <h4>Title: {titles[0]}</h4>}
      {aliases.length > 1 && <ul>Aliases: {aliasElements}</ul>}
    </section>
  );};

Character.propTypes = {
  name: PropTypes.string,
  culture: PropTypes.string,
  titles: PropTypes.array,
  aliases: PropTypes.array
};

export default Character;
