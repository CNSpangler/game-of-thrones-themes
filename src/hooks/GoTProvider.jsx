import React, { createContext, useContext, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { fetchCharacters } from '../../services/GoT-API.js';

const GoTContext = createContext();

// wrapper to provide state to child components
export const GoTProvider = ({ children }) => {
  const [characters, setCharacters] = useState([]);
  const [page, setPage] = useState(6);

  useEffect(() => {
    fetchCharacters(page)
      .then(results => setCharacters(results));
  }, [page]);

  return (
    <GoTContext.Provider value={{ characters, page }}>
      {children}
    </GoTContext.Provider>
  );
};

GoTProvider.propTypes = {
  children: PropTypes.node
};

export const useCharacters = () => {
  const { characters } = useContext(GoTContext);
  return characters;
};

// export const useToggle = () => {
//   const { toggle } = useContext(GoTContext);
//   return toggle;
// };
