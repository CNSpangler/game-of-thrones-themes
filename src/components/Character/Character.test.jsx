import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';
import { GoTProvider } from '../../hooks/GoTProvider';

describe('Character component', () => {
  it('renders Character', () => {
    const wrapper = shallow(
      <GoTProvider>
        <Character />
      </GoTProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
