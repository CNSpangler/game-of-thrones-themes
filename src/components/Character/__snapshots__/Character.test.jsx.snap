// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<ContextProvider
  value={
    Object {
      "characters": Array [],
      "fireIce": "fire",
      "page": 6,
      "toggle": [Function],
      "updatePage": [Function],
    }
  }
>
  <Character />
</ContextProvider>
`;
